validation/markets-validator.js

```diff
@@ -464,6 +479,31 @@ function MetadataValidator(metadata) {
             this.errors.push(new NullAttributeIssue("url"));
         } else if (url.length === 0) {
             this.errors.push(new EmptyAttributeIssue("url"));
+        } else {
+            url = modurl.parse(url);
+            var request;
+            switch(url.protocol) {
+                case 'http:': request = http; break;
+                case 'https:': request = https; break;
+                default:
+                    this.warnings.push(new CustomIssue('Unknown protocol "' + url.protocol + '" in data source url.'));
+            }
+            if (request) {
+                url.method = 'HEAD';
+                url.timeout = 10000; // 10 seconds
+                request = request.request(url, function (response) {
+                    if (response.statusCode !== 200)
+                        asyncWarnings.push(new CustomIssue(cityName + ': HTTP response status of data source url was ' + response.statusCode + '.'));
```

Comment:
Some of these lines are rather long for my taste, however I'm not sure whether we have decided on a common policy, yet.